OOP-Assignment1-Charles/OOP-Assignment1-Charles/Analyse.cs

```diff
@@ -6,31 +6,193 @@
-        public List<int> analyseText(string input)
+        List<string> vowel = new List<string>();
+
+        public void WordFrequency(string text)
         {
+
+            int[] value = new int[(int)char.MaxValue];
+
+            foreach (char t in text)
+            {
+                value[(int)t]++;
+            }
+
+            // Outputs all letters and their respective frequecncies
+            for (int i = 0; i < (int)char.MaxValue; i++)
+            {
+                if (value[i] > 0 && char.IsLetterOrDigit((char)i))
+                {
+                    Console.WriteLine($"Letter: {(char)i}  Frequency: {value[i]}");
+                }
+
+            }
+
+
+        }
+
+        public List<int> analyseText(string[] input)
+        {
+            string toAnalyse = input[0].ToString(); // toAnalyse stores the users inputted text
+            string fullStop = ".";
+            char[] vowels = { 'a', 'e', 'i', 'o', 'u' }; //This would preferably be tuple, however this causes indexing issues so a normal aray has to be usedl
+            char[] consonants = { 'b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 'r', 's', 't', 'v', 'x', 'z' };
+            string[] alphabet = {"a", "b", "c", "d", "e", "f", //String because c# doesnt operate required functions of char elements
+                "g", "h", "i", "j", "k", "l", "m", "n", "o",
+                "p", "q", "r", "s" , "t", "u", "v", "w", "x", "w", "z"};
+
+            /*string[] alphabetHigher = alphabetLower;
+            alphabetHigher = Array.ConvertAll(alphabetHigher, d => d.ToLower());*/
+
+
+            int totalSentances = 0; // A list could preferably be used here to save time. Although variables
+            int totalCharacters = 0; // are more readable from first glance and increase the readibility of the program.
+            int totalVowels = 0;     // For example, if items[0] was used in a instuction it would be unclear what its purpose is,
+            int totalconsonants = 0;     // For example, if items[0] was used in a instuction it would be unclear what its purpose is,
+            int totalUpper = 0;       // and perhaps cause some confusion
+            int totalLower = 0;
+
+
             //List of integers to hold the first five measurements:
             //1. Number of sentences
+
+
+
+            foreach (var item in toAnalyse)
+            {
+
+                if (item.ToString() == fullStop)
+                {
+                    totalSentances++;// When a full stop is found. The sentacne count increments by one.
+                }
+
+            }
+
+
             //2. Number of vowels
+
+
+
+
+            for (int i = 0; i < vowels.Length; i++)
+            {
+                for (int j = 0; j < toAnalyse.Length; j++)
+                {
+                    if (vowels[i] == toAnalyse[j])
+                    {
+                        totalVowels++;
+                    }
+                }
+            }
```

Comment:
Might be more readable if you use foreach and search through them instead getting of length?